corehq/apps/hqadmin/templates/hqadmin/app_build_timings.html

```diff
@@ -0,0 +1,47 @@
+{% extends "hqwebapp/base.html" %}
+{% load compress %}
+{% load hq_shared_tags %}
+
+{% block title %}App Build Times{% endblock %}
+
+{% block stylesheets %}{{ block.super }}
+    <link type="text/css" rel="stylesheet" href="{% static 'jquery-treetable/css/jquery.treetable.css' %}"/>
+{% endblock stylesheets %}
+
+{% requirejs_main 'hqadmin/js/app_build_timings' %}
+
+{% block content %}
+<div class="container-fluid">
+    <h1>App Build Timings</h1>
+    <h4>
+        <a href="{% url 'view_app' app.domain app.master_id %}">
+            {{ app.name }}
+        </a>
+        <small>{{ app.master_id }}</small>
+    </h4>
+
+    <div class="row">
+        <table id="timingTable" class="table">
+            <thead>
+                <tr>
+                    <th>Timer name</th>
+                    <th>Duration</th>
+                    <th>Percent of Parent</th>
+                    <th>Percent of Total</th>
+                </tr>
+            </thead>
+            <tbody>
+                {% for timer in timing_data %}
+                    <tr data-tt-id="{{ timer.name }}" {% if timer.parent %}data-tt-parent-id="{{ timer.parent.name }}"{% endif %}>
+                        <td>{{ timer.name }}</td>
+                        <td>{{ timer.duration|stringformat:".3f" }}</td>
+                        <td>{{ timer.percent_of_parent|stringformat:".1f" }}</td>
+                        <td>{{ timer.percent_of_total|stringformat:".1f" }}</td>
+                    </tr>
+                {% endfor %}
+            </tbody>
+        </table>
+    </div>
```

Comment:
This block is shared with the admin restore view - is there a simple way to pull it out along with the associated css and js?

Reply:
You could pull the table element out as a partial and pass `timing_data` as a variable to it.

For the js, you could use `hqadmin/js/admin_restore` as the main module for both of these (probably renaming it and definitely wrapping the `baseAce.initAceEditor` call in a conditional that checks if the page has a `#payload` element). 

Alternatively, there's some precedent for having app-specific modules that just exist to initialize widgets, like [accounting/js/widgets](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/accounting/static/accounting/js/widgets.js#L133-L136). So you could rename `hqadmin/js/app_build_timings` to `hqadmin/js/widgets`, leave the code in it alone (maybe tweak it to use a more generic class name instead of `#timingTable`), and then add `hqadmin/js/widgets` as a dependency to `hqadmin/js/admin_restore`.

We don't have a good way of bundling the js+css and sharing them as one unit. We used to do this with django partial templates that just consisted of script and link tags, but with requirejs we're no longer using script tags. There are some options [like this](https://github.com/guybedford/require-css) for integrating CSS dependencies with requirejs, but I haven't dug into them (it's also very possible that we'll move off of requirejs and onto something like webpack and use that to manage CSS).

Reply:
Pulled out the template as an includes and left a comment at the top about usage:
https://github.com/dimagi/commcare-hq/pull/22181/commits/d05985548d59705aa91aad453c981becd7f28a51